src/Analyzers/CSharp/Analyzers/UseCollectionExpression/CSharpUseCollectionExpressionForStackAllocDiagnosticAnalyzer.cs

```diff
@@ -158,94 +154,11 @@ private static void AnalyzeExplicitStackAllocExpression(SyntaxNodeAnalysisContex
         StackAllocArrayCreationExpressionSyntax expression,
         CancellationToken cancellationToken)
     {
-        // has to either be `stackalloc X[]` or `stackalloc X[const]`.
```

Comment:
moved to common helper code, since we'll use it for stackalloc *and* new array creations.